src/univalue/include/univalue.h

```diff
@@ -174,6 +174,15 @@ class UniValue {
     const UniValue& get_obj() const;
     const UniValue& get_array() const;
 
+    bool get(bool default_value) const                             { return isNull() ? default_value : get_bool(); }
+    const std::string& get(const std::string& default_value) const { return isNull() ? default_value : get_str(); }
```

Comment:
```suggestion
    std::string get(const std::string& default_value) const { return isNull() ? default_value : get_str(); }
```

This will need to return a copy. Otherwise we'll get memory issues.

Reply:
`get_str()` also returns a `const std::string&`, so it should be ok, no?

Reply:
Its the textbook example from https://youtu.be/lkgszkPnV8g?t=845 ;)

Reply:
Huh... that looks like a bug in the C++ spec, tbh. Anyway, fixed.

Reply:
> Its the textbook example from [youtu.be/lkgszkPnV8g?t=845](https://youtu.be/lkgszkPnV8g?t=845) ;)

Thanks for the good video link :+1:

Reply:
Yup cool stuff.